keps/sig-apps/20190715-rollingUpdate-enhancement-for-daemonset.md

```diff
@@ -74,13 +74,15 @@ Consider the following scenarios:-
 - Add new fields `Selector` and `Partition` for existing RollingUpdateDaemonSet.
 
 ### Non-Goals
-- This inplement is only implemeted to affect to update strategy of Daemonset. No other behaviors of Daemonset will be changed.
+- This implement is only implemeted to affect to update strategy of Daemonset. No other behaviors of Daemonset will be changed.
 
 ## Proposal
 
 ### User Stories
 
 #### Story 1
+In some end users' clusters, they deploy DNS resolver and other containers serving infrastructure by daemonset in every nodes. To an online e-commercial services which serve 200K+ QPS, it must cause cascading disaster to stop those infrastructure services even in a very short time by current Daemonset RollingUpdate strategy. So surgingRollingUpdate is a necessary strategy for our production environment. 
```

Comment:
I don't think a rolling update would solve this problem, you can't replace a running pod which specify host port in the spec without stopping it. `SurgingRollingUpdate` does not make sense to me, delete by partitions or in specific order may mitigate this problem.

Reply:
SurgingRollingUpdate will solve those cases without binding host port, so it could be a supplement for existing update strategy. Operators should have knowledge to choose suitable one. As to partitions and selector, they can help to avoid mistake and reduce its cost.

Reply:
@draveness Not sure if I'm understanding your correctly but: how rollout strategy is related to host port?

Reply:
> @draveness Not sure if I'm understanding your correctly but: how rollout strategy is related to host port?

It is a common pattern to use host port in daemonset, suppose we use a rolling update strategy like deployment, deamonsets should create a new pod with the same hostport on the same machine, but it will fail to start since the port on the machine is owned by the older daemonset pod. SurgingRollingUpdate does not hold since we can't create more pods with the same port on the same machine. 

> SurgingRollingUpdate will solve those cases without binding host port, so it could be a supplement for existing update strategy. 

If we use a damonset without hostport, why not using deployment instead.

Reply:
We could use daemonset for log collecting which does not need a hostport, but I think two log collecting pod running on the same machine could be problematic. Add a `maxUnavailable` field instead of surging to the daemonset pod would make more sense.

Reply:
I don't think that's the intention of this proposal which is trying to propose an optional strategy instead of enhancement for existing ones. 

But yeah, this KEP dose not make sense to me as well since it mixed too many things together, and the use cases are not clear.